test/org/zaproxy/zap/extension/pscanrulesBeta/InformationDisclosureDebugErrorsUnitTest.java

```diff
@@ -0,0 +1,296 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright 2016 The ZAP Development Team
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.pscanrulesBeta;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+
+public class InformationDisclosureDebugErrorsUnitTest extends PassiveScannerTest {
+	private static final String URI = "https://www.example.com/";
+	private static final String defaultErrorMessage = "Internal Server Error";
+	private static final List<String> debugErrors = Arrays.asList(
+			defaultErrorMessage,
+			"There seems to have been a problem with the",
+			"This error page might contain sensitive information because ASP.NET",
+			"PHP Error"
+			);
+	
+	@Rule
+    public TemporaryFolder testFolder = new TemporaryFolder();
+	
+	@Override
+	protected InformationDisclosureDebugErrors createScanner() {
+		InformationDisclosureDebugErrors scanner = new InformationDisclosureDebugErrors();
+		
+		try {
+			Path errorFile = testFolder.newFile("debug-error-messages.txt").toPath();
+			Files.write(errorFile, debugErrors, Charset.forName("UTF-8"));
+			scanner.setDebugErrorFile(errorFile);
+		} catch (IOException e) {
+			
+		}
+		
+		return scanner;
+	}
+	
+	@Test
+	public void alertsIfDebugErrorsDisclosed() throws HttpMalformedHeaderException {
+		for (int i = 0; i < debugErrors.size(); i++) {
+			String debugError = debugErrors.get(i);
+			String responseBody = "<html>" + debugError + "</html>"; 
+			
+			HttpMessage msg = new HttpMessage();

```

Comment:
The creation of the messages could be extracted to a method, since it's (almost) always the same. Something like `createHttpMessageWithRespBody(responseBody)`.